.github/copilot-instructions.md

```diff
@@ -0,0 +1,220 @@
+# GitHub Copilot Development Environment Instructions
+
+This document provides specific guidance for GitHub Copilot when working on the .NET MAUI repository. It serves as context for understanding the project structure, development workflow, and best practices.
+
+## Repository Overview
+
+**.NET MAUI** is a cross-platform framework for creating mobile and desktop applications with C# and XAML. This repository contains the core framework code that enables development for Android, iOS, iPadOS, macOS, and Windows from a single shared codebase.
+
+### Key Technologies
+- **.NET SDK** - Version depends on the branch:
+  - **main branch**: Use the latest stable version of .NET to build
+  - **net10.0 branch**: Use the latest .NET 10 SDK
+  - **etc.**: Each feature branch correlates to its respective .NET version
+- **C#** and **XAML** for application development
+- **Cake build system** for compilation and packaging
+- **MSBuild** with custom build tasks
+- **xUnit** for testing
+
+## Development Environment Setup
+
+### Prerequisites
+
+#### Linux Development (Current Environment)
+For .NET installation on Linux, follow the official Microsoft documentation:
+* https://learn.microsoft.com/en-us/dotnet/core/install/linux
+
+#### Additional Requirements
+- **OpenJDK 17** for Android development
+- **VS Code** with .NET MAUI Dev Kit extension
+- **Android SDK** for Android development
+
+### Initial Repository Setup
+
+1. **Clone and navigate to repository:**
+   ```bash
+   git clone https://github.com/dotnet/maui.git
+   cd maui
+   ```
+
+2. **Restore tools and build tasks (REQUIRED before opening IDE):**
+   ```bash
+   dotnet tool restore
+   dotnet build ./Microsoft.Maui.BuildTasks.slnf
+   ```
+
+## Project Structure
+
+### Important Directories
+- `src/Core/` - Core MAUI framework code
+- `src/Controls/` - UI controls and components
+- `src/Essentials/` - Platform APIs and essentials
+- `src/TestUtils/` - Testing utilities and infrastructure
+- `docs/` - Development documentation
+- `eng/` - Build engineering and tooling
+- `.github/` - GitHub workflows and configuration
+
+### Platform-Specific Code Organization
```

Comment:
Add notes here about how any files with

".windows.cs" will only compile for the windows TFM
".android.cs" will only compile for the android TFM
".ios.cs" will only compile for the iOS and MacCatalyst TFM
"MacCatalyst.cs" will only compile for the MacCatalyst TFM